src/components/popover/Popover.tsx

```diff
@@ -1,151 +1,277 @@
+    placement,
+  });
 
-  const defaultOnToggle = React.useCallback((newOpen: boolean | ((prevState: boolean) => boolean)) => {
-    setOpen(newOpen);
-  }, []);
+  const id = useId();
+  const labelId = `${id}-label`;
+  const descriptionId = `${id}-description`;
+
+  const interactions = [useClick(context), useRole(context), useDismiss(context, { ...dismissOptions })];
+
+  const { getReferenceProps, getFloatingProps } = useInteractions(
+    hoverable ? [...interactions, useHover(context)] : interactions
+  );
 
   React.useEffect(() => {
-    if (!init) {
-      if ('current' in boundaryElement && boundaryElement.current) {
-        setInit(true);
-      }
+    if (refs.reference.current && refs.floating.current && isOpen) {
+      return autoUpdate(
+        refs.reference.current,
+        refs.floating.current,
+        update,
+        updateOptions ? { ...updateOptions } : { ancestorScroll: false }
+      );
     }
-  }, [boundaryElement]);
+    return;
+  }, [isOpen, update, refs.reference, refs.floating]);
+
+  React.useEffect(() => {
+    onToggle && onToggle(context);
+  }, [context]);
 
   const classes = classNames(
     {
-      Popover: true,
       ['Popover--dark']: dark,
+      ['mt-4']: true,
     },
-    className
+    `Popover ${className}`
   );
 
-  const PopoverWrapper = (
-    <div data-test="DesignSystem-Popover" className={classes} data-layer={true}>
-      {children}
-    </div>
-  );
+  const positionOffset = {
+    'auto-start': 'top left',
+    auto: 'top',
+    'auto-end': 'top right',
+    'top-start': 'bottom left',
+    top: 'bottom',
+    'top-end': 'bottom right',
+    'right-start': 'top right',
+    right: 'top right',
+    'right-end': 'top right',
+    'bottom-end': 'top right',
+    bottom: 'top',
+    'bottom-start': 'top left',
+    'left-end': 'top left',
+    left: 'top left',
+    'left-start': 'top left',
+  };
 
   return (
-    <PopperWrapper
-      {...rest}
-      init={init}
-      boundaryElement={'current' in boundaryElement ? boundaryElement.current : boundaryElement}
-      open={open}
-      hide={hideOnReferenceEscape}
-      style={customStyle}
-      onToggle={onToggle || defaultOnToggle}
-      placement={position}
-    >
-      {PopoverWrapper}
-    </PopperWrapper>
+    <>
+      {React.cloneElement(trigger, {
+        ref: reference,
+        ...getReferenceProps({ ref: reference }),
+        // Need to set onClick here to handle open=true on initial state
+        onClick: () => customSetOpen(),
+      })}
+      <FloatingPortal>
+        <AnimatePresence>
+          {isOpen && (
+            <FloatingFocusManager context={context}>
+              <motion.div
+                data-test="DesignSystem-Popover"
+                initial={{ opacity: 0.5, scaleY: 0 }}
+                animate={{
+                  opacity: 1,
+                  scaleY: 1,
+                  animationTimingFunction: isOpen ? 'cubic-bezier(0, 0, 0.38, 0.9)' : 'cubic-bezier(0.2, 0, 1, 0.9)',
```

Comment:
shouldn't we store all these `cubic-bezier` curves in the global CSS class and then reuse them? Please refer this https://design-dev.innovaccer.com/foundations/interactions/#exit-easing